src/components/FilterSearch.tsx

```diff
@@ -77,23 +78,39 @@ export function FilterSearch({
     return { ...searchField, fetchEntities: false };
   });
   const cssClasses = useComposedCssClasses(builtInCssClasses, customCssClasses);
+  const [currentFilter, setCurrentFilter] = useState<Filter>();
+  const [query, setQuery] = useState<string>();
+  const filters = useSearchState(state => state.filters.static);
 
   const [
     filterSearchResponse,
-    executeFilterSearch
+    executeFilterSearch,
+    clearFilterSearchResponse
   ] = useSynchronizedRequest<string, FilterSearchResponse>(
-    inputValue => searchActions.executeFilterSearch(inputValue ?? '', sectioned, searchParamFields),
+    inputValue => {
+      setQuery(inputValue);
+      return searchActions.executeFilterSearch(inputValue ?? '', sectioned, searchParamFields);
+    },
     (e) => console.error('Error occured executing a filter search request.\n', e)
   );
 
+  useEffect(() => {
+    filters?.forEach(f => {
+      if (currentFilter && isDuplicateFilter(f, currentFilter) && !f.selected) {
```

Comment:
Is it checking if some other component unchecks a static filter with the same filter?

Reply:
maybe instead of a .forEach() this can be a .find()? or something else that short circuits, since theres no point of calling this multiple times in a row

Reply:
yup. If AppliedFilters deselect a filter that is from a FilterSearch, it should reset the corresponding FilterSearch component's state (clear filterSearchResponse data, input text, and `currrentFilter` state). Sure, I can refactor this to use find()